storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -103,9 +138,12 @@ public void write(
         if (overwrite) {
             writeActions(fs, path, actions);
             return;
+        } else if (fs.exists(path)) {
```

Comment:
This check is not needed. Right? Are you just trying to avoid touching DynamoDBLogStore if the file already exists?

Reply:
Gotcha. I just wanted to confirm this is about optimization rather than correctness.

Reply:
Gotcha. I just wanted to confirm this is about optimization rather than correctness.